### PR TITLE
Inventory system cleanup + More quick equip keybinds.

### PR DESCRIPTION
**Pull request by Xander3359**
## About The Pull Request
Making this PR as finished before I get too ambitious.
So what have I done exactly?
Changed the UI a bit in game pref to allow for 5 quick equips.
Added 5 keybinds in settings to utilize those 5 quick equips. (Only 2 are set by default)
Added some slot defines in equipment.dm
Fix some doc in storage.dm
The amount of headache caused from TGUI is making me insane.
image
image

## Why It's Good For The Game
General improvements are good. 5 quick_equip keybinds that you can choose which slot to draw from is really nice and makes the game much more customizable.
General code cleanup is good to do as well. Fucky spacing, things that aren't properly commented. etc. Nothing major.
## Changelog
:cl:
add: There are now 5 quick equip keybinds.
del: preferred_slot client verb as it has become unecessary.
code: Some cleanup in a few files I touched upon, nothing major.
/:cl:

### CODE REVIEW

File: code/datums/keybinding/human.dm [hunk 1]
```diff
@@ -1,65 +1,76 @@
...
-	description = ""
 	keybind_signal = COMSIG_KB_QUICKEQUIP
+	///The quick equip slot the keybind will equip to, checked by the reciever of the keybind signal.
+	var/quick_equip_slot
+
+/datum/keybinding/human/quick_equip/quick_equip_1
+	hotkey_keys = list("E")
+	name = "quick_equip_1"
+	full_name = "Quick equip 1"
+	quick_equip_slot = 1
```
Comment (TiviPlus) [inline code/datums/keybinding/human.dm:16]: You can just use a macro here to minimize boilerplate
Also no numbers in typepaths

File: code/modules/client/preferences.dm [hunk 2]
```diff
@@ -79,8 +79,8 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 	///Your preferred squad, when playing SOM
 	var/preferred_squad_som = "None"
 	var/alternate_option = RETURN_TO_LOBBY
-	var/preferred_slot = SLOT_S_STORE
-	var/preferred_slot_alt = SLOT_BACK
+	///Quick equip slots, up to 5. One per quick equip hotkey.
+	var/list/quick_equip = list()
```
Comment (TiviPlus) [inline code/modules/client/preferences.dm:83]: Up to [DEFINE] not 5

File: code/modules/client/preferences_savefile.dm [hunk 3]
```diff
@@ -174,46 +174,47 @@
...
+	custom_emotes = sanitize_is_full_emote_list(custom_emotes)
+	chem_macros = sanitize_islist(chem_macros, list())
+	quick_equip = sanitize_islist(quick_equip, QUICK_EQUIP_ORDER, MAX_QUICK_EQUIP_SLOTS)
 
-	mute_self_combat_messages	= sanitize_integer(mute_self_combat_messages, FALSE, TRUE, initial(mute_self_combat_messages))
-	mute_others_combat_messages	= sanitize_integer(mute_others_combat_messages, FALSE, TRUE, initial(mute_others_combat_messages))
-	mute_xeno_health_alert_messages	= sanitize_integer(mute_xeno_health_alert_messages, FALSE, TRUE, initial(mute_xeno_health_alert_messages))
+	mute_self_combat_messages = sanitize_integer(mute_self_combat_messages, FALSE, TRUE, initial(mute_self_combat_messages))
+	mute_others_combat_messages = sanitize_integer(mute_others_combat_messages, FALSE, TRUE, initial(mute_others_combat_messages))
+	mute_xeno_health_alert_messages = sanitize_integer(mute_xeno_health_alert_messages, FALSE, TRUE, initial(mute_xeno_health_alert_messages))
```
Comment (TiviPlus) [inline code/modules/client/preferences_savefile.dm:212]: Please atomize stuff like this in future it just makes it harder to review

File: code/modules/client/preferences_ui.dm [hunk 4]
```diff
@@ -17,126 +17,128 @@
 	return GLOB.always_state
 
 /datum/preferences/ui_data(mob/user)
-	. = list()
-	.["tabIndex"] = tab_index
-	.["slot"] = default_slot
-	.["save_slot_names"] = list()
+	var/list/data = list()
+	data["tabIndex"] = tab_index
```
Comment (TiviPlus) [inline code/modules/client/preferences_ui.dm:21]: Ditto

File: tgui/packages/tgui/interfaces/PlayerPreferences/GameSettings.tsx [hunk 5]
```diff
@@ -250,16 +251,18 @@ export const GameSettings = (props, context) => {
...
+                  <Box>Quick equip #{index_slot + 1}</Box>
+                  <Button
+                    key={equip_slot}
+                    content={equip_slot}
+                    onClick={() =>
+                      act('change_quick_equip', { selection: index_slot + 1 })
+                    }
+                  />
+                </>
+              ))}
```
Comment (TiviPlus) [inline tgui/packages/tgui/interfaces/PlayerPreferences/GameSettings.tsx:265]: This shouldve been split into its own component but whatever

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/client/preferences_savefile.dm (hunk 3 above)
Comment (Xander3359) [reply code/modules/client/preferences_savefile.dm:212]: Will do, I just got over ambitious when I was looking at inventory code when the idea popped in my head.

**Pull request merged**